### PR TITLE
refactor: Migrate Play Page and Lobby Functions to Socket

### PR DESCRIPTION
**Pull request by ryxavyer**
Migrated public lobby query and create / join / message functions of lobby.

### CODE REVIEW

File: server/server.js [hunk 1]
```diff
@@ -20,6 +28,134 @@ app.get('/', (req, res) => {
...
+            inGame: false,
+            isPrivate: false,
+            gameID: null,
+            users: {},
+            messages: [],
+            createdAt: new Date().toISOString(),
+        });
+        socket.emit('lobbyCreated', { lobbyCode: newLobbyCode });
+    });
+    socket.on('joinLobby', async (lobbyCode, player) => {
```
Comment (Klaytee22) [inline server/server.js:73]: we'll need to look into adding logic for things like full lobbies. Ideally we have this logic on server as well as on the db as a db rule

File: server/server.js [hunk 2]
```diff
@@ -20,6 +28,134 @@ app.get('/', (req, res) => {
...
+        );
+        io.to(lobbyCode).emit('lobbyMessages', { messages: lobby.messages });
+    });
+    socket.on('updateReady', async (lobbyCode, player, newStatus) => {
+        const lobby = await database.collection(lobbyCollectionName).findOneAndUpdate(
+            { code: lobbyCode },
+            { $set: { [`users.${player.uid}.ready`]: newStatus } },
+            { returnDocument: "after"}
+        );
+        socket.emit('readyUpdated', { newStatus })
```
Comment (Klaytee22) [inline server/server.js:108]: shouldn't need this value since it shouldn't ever be different than the one in the room's lobbyReadyUpdated table

**Review by Klaytee22 — APPROVED** [review]

File: server/server.js (hunk 1 above)
Comment (ryxavyer) [reply server/server.js:72]: updated

File: server/server.js (hunk 2 above)
Comment (ryxavyer) [reply server/server.js]: updated

**Review by Klaytee22 — APPROVED** [review]

**Pull request merged**